ChargedUp/src/main/java/com/gos/chargedup/subsystems/ArmPivotSubsystem.java

```diff
@@ -0,0 +1,534 @@
+                .addP(0.0045) // 0.0058
+                .addD(0.045)
+                .build();
+        }
+    }
+
+    @Override
+    public void periodic() {
+        m_sparkPidProperties.updateIfChanged();
+        m_wpiPidProperties.updateIfChanged();
+        m_wpiFeedForward.updateIfChanged();
+
+        m_lowerLimitSwitchEntry.setBoolean(isLowerLimitSwitchedPressed());
+        m_upperLimitSwitchEntry.setBoolean(isUpperLimitSwitchedPressed());
+        m_encoderDegEntry.setNumber(getArmAngleDeg2());
+        m_absoluteEncoderEntry.setNumber(getAbsoluteEncoderAngle2());
+        m_absoluteEncoderToBuiltInDriftEntry.setNumber(getAbsoluteEncoderAngle2() - getArmAngleDeg2());
+        m_goalAngleDegEntry.setNumber(m_armAngleGoal);
+        m_velocityEntry.setNumber(getArmVelocityDegPerSec2());
+
+        m_pivotErrorAlert.checkAlerts();
+    }
+
+    @Override
+    public void simulationPeriodic() {
+        m_pivotSimulator.update();
+    }
+
+    public void pivotArmUp() {
+        m_pivotMotor.set(ARM_MOTOR_SPEED);
+    }
+
+    public void pivotArmDown() {
+        m_pivotMotor.set(-ARM_MOTOR_SPEED);
+    }
+
+    public double getArmMotorSpeed() {
+        return m_pivotMotor.getAppliedOutput();
+    }
+
+    public void pivotArmStop() {
+        m_armAngleGoal = Double.MIN_VALUE;
+        m_pivotMotor.set(0);
+    }
+
+    public final double getArmAngleDeg2() {
+        return m_pivotMotorEncoder.getPosition();
+    }
+
+    public double getArmVelocityDegPerSec2() {
+        return m_pivotMotorEncoder.getVelocity();
+    }
+
+    public double getArmAngleGoal() {
+        return m_armAngleGoal;
+    }
+
+    public boolean isUpperLimitSwitchedPressed() {
+        return !m_upperLimitSwitch.get();
+    }
+
+    public boolean isLowerLimitSwitchedPressed() {
+        return !m_lowerLimitSwitch.get();
+    }
+
+    private boolean isMotionProfileFinished() {
+        return m_wpiPidController.getSetpoint().equals(m_wpiPidController.getGoal());
+    }
+
+    public void pivotArmToAngle2(double pivotAngleGoal) {
+        m_armAngleGoal = pivotAngleGoal;
+
+        m_wpiPidController.calculate(getFeedbackAngleDeg(), pivotAngleGoal);
+
+        TrapezoidProfile.State profileSetpointDegrees = m_wpiPidController.getSetpoint();
+        m_profileVelocityGoalEntry.setNumber(profileSetpointDegrees.velocity);
+        m_profilePositionGoalEntry.setNumber(profileSetpointDegrees.position);
+
+        double feedForwardVolts = m_wpiFeedForward.calculate(
+                Units.degreesToRadians(profileSetpointDegrees.position),
+                Units.degreesToRadians(profileSetpointDegrees.velocity));
+        m_pidArbitraryFeedForwardEntry.setNumber(feedForwardVolts);
+
+        if (isMotionProfileFinished()) {
+            m_sparkPidController.setReference(pivotAngleGoal, CANSparkMax.ControlType.kPosition, 0, feedForwardVolts);
+        } else {
+            m_pivotMotor.setVoltage(feedForwardVolts);
+        }
```

Comment:
HACK ALERT 🚨 

We probably should have just gone with traditional PID. I don't like that the WPI profiled version does not recalculate, so if you fall behind you stay behind.